src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -243,49 +237,48 @@ private bool StartCore(ProcessStartInfo startInfo)
                 }
             }
 
-            int childPid = -1, stdinFd = -1, stdoutFd = -1, stderrFd = -1, result = -1;
+            int childPid, stdinFd, stdoutFd, stderrFd;
             string[] envp = CreateEnvp(startInfo);
             string cwd = !string.IsNullOrWhiteSpace(startInfo.WorkingDirectory) ? startInfo.WorkingDirectory : null;
 
-            filename = ResolvePath(startInfo.FileName); 
-            if (!string.IsNullOrEmpty(filename))
-            {
-                argv = ParseArgv(startInfo);
-
-                // Invoke the shim fork/execve routine.  It will create pipes for all requested
-                // redirects, fork a child process, map the pipe ends onto the appropriate stdin/stdout/stderr
-                // descriptors, and execve to execute the requested process.  The shim implementation
-                // is used to fork/execve as executing managed code in a forked process is not safe (only
-                // the calling thread will transfer, thread IDs aren't stable across the fork, etc.)
-                result = Interop.Sys.ForkAndExecProcess(
-                    filename, argv, envp, cwd,
-                    startInfo.RedirectStandardInput, startInfo.RedirectStandardOutput, startInfo.RedirectStandardError,
-                    out childPid,
-                    out stdinFd, out stdoutFd, out stderrFd, shouldThrow: !startInfo.UseShellExecute);
-            }
+            filename = ResolvePath(startInfo.FileName);
+            argv = ParseArgv(startInfo);
 
-            if (result != 0)
+            if (!startInfo.UseShellExecute)
             {
-                if (!startInfo.UseShellExecute)
+                if (string.IsNullOrEmpty(filename))
                 {
-                    // Could not find the file
-                    Debug.Assert(string.IsNullOrEmpty(filename));
                     throw new Win32Exception(Interop.Error.ENOENT.Info().RawErrno);
                 }
 
-                // this time, set the filename as default program to open file/url
+                if (Directory.Exists(startInfo.FileName))
+                {
+                    throw new Win32Exception(SR.DirectoryNotValidAsInput);
+                }
+            }
+            else
+            {
+                if (string.IsNullOrEmpty(filename))
+                {
+                    filename = startInfo.FileName; // is url
```

Comment:
Is this necessary? `filename` is overwritten in the next statement anyway. Correct me if I'm wrong.